File: UI/Assignment8/Queue.js
var queue = function() {
    this.array = [];
    this.enqueue = function(element) {
        this.array.push(element);
    }
    this.dequeue = function() {
        if (this.array.length == 0) {
            console.log("Queue is empty");
        } else {
            this.array.shift();
        }
    }
}


list = new queue();
list.enqueue(20);
list.enqueue(40);
list.array;
list.dequeue();
list.array;

File: DSA/Assignment2/DSA2/src/counseling/Queue.java
package counseling;

public class Queue{
    int rear, front;
    int size;
    String queue[];


    Queue(int arraySize) {
	    queue = new String[arraySize];
	}

	public boolean isFull() {
		if (size == queue.length)
			return true;
		return false;
	}
	
	public void enqueue(String data) {
		if (isFull())
			System.out.println("queue is full");
		else{
			size++;
			front = front%(queue.length);
			queue[front++] = data;
		}
	}

	public String dequeue() {
		if (isEmpty()){
			return "queue is empty";
		}
		else{
			size--;
			return "Dequeue "+queue[rear++];
		}
	}
	
	private boolean isEmpty() {
		if (size == 0)
			return true;
		return false;
	}

	public void display() {
		if (front <= rear){
			for (int index=rear; index<queue.length; index++)
				System.out.print(queue[index]+", ");
			for (int index=0; index!=front; index++)
				System.out.print(queue[index]+", ");
		}
		else
			for (int indexi=rear; indexi<front; indexi++)
				System.out.print(queue[indexi]+", ");
		System.out.println();
	}


    public String[] getQueue() {
        return queue;
    }
}
